packages/SwingSet/docs/vat-environment.md

```diff
@@ -0,0 +1,113 @@
+
+SES is a safe deterministic subset of "strict mode" JavaScript, which means it does not include any of these IO objects (they would provide ambient authority, which is not "safe"). It also modifies a few built-in objects to remove non-determinism.
+
+The full list of SES globals (as of SES-0.8.0) is defined in [the SES source code](https://github.com/Agoric/SES-shim/blob/SES-v0.8.0/packages/ses/src/whitelist.js). It includes the core language objects: `Object`, `Array`, `Number`, `Map` and `WeakMap`, `Number`, `BigInt`, and others. This is a subset of the globals defined by the baseline JavaScript language specification.
+
+This includes `Math`, but `Math.random` is disabled (calling it throws an error) as an obvious source of non-determinism. Likewise `Date` is included, but `new Date()` and `Date.now` throw an error. The remaining `Math` and `Date` features are purely computational and deterministic.
+
+Much of the `Intl` package, and some locale-specifc part of other objects (e.g. `Number.prototype.toLocaleString`) has results which depend upon which locale is configured, and that will vary from one process to another. The way we handle this is still in development. Either these functions will be disabled, or they will act as if they run on a host with a single fixed locale defined by the SES specification.
+
+As SES is on the JavaScript standards track, it includes other proposed features. If those features become standards, future JS environments will include them as global objects. So the current SES shim also makes those global objects available.
+
+## Additions
+
+`console` is available to vat code as a deliberate addition. It is not part of JavaScript, but it's so important for debugging that leaving it out would cause too much confusion. Note that the exact behavior of `console.log` is up to the host program: it is not guaranteed to be displayed to the operator. Applications that want to let vats write to `process.stdout` must somehow provide that authority directly. From a vat's point of view, `console` is frozen and powerless.
+
+`harden` is available as a global. It freezes the API surface (enumerable data properties) of an object. For more details, see [the `harden` package](https://github.com/Agoric/SES-shim/blob/SES-v0.8.0/packages/harden/README.md).
+
+`HandledPromise` is also a global. The `E` wrapper (`E(target).methodname(args)`) can be imported as `import { E } from '@agoric/eventual-send`. These are defined by the TC39 [Eventual-Send Proposal](https://github.com/tc39/proposal-eventual-send). In addition, "tildot" syntax (`target~.methodname(args)`) can be used by vat code.
```

Comment:
If I'm reading this right, "These two are..." for "These are"; it took me a moment to realize "these" was referring to HandledPromise and the E wrapper (which is also a really bad name for a Springsteen cover band....: -))

Reply:
hah! yup, that's right. will apply.